### ISSUE TITLE
MC Dropout on decoder layer

### ISSUE THREAD

**Issue by jmaxrdgz** [open-mmlab/mmdetection#12346]
Hi, I am trying to implement Monte Carlo Dropout on decoder layers of DINO using MMDetection.

Does anyone has clues on how to activate dropout with a specific value, different then the default "dropout" used at training time and specified in the config file, to be used at inference time ?

**Comment by jihu-amkor** [comment]
Hi, i was tryied MC at the inference phase too.

For dropout activate on inference, i implemented like this.

implemented custom classes of inferencer.py, load_model.py

on the load_model.py, used DetInferencer, i implemented activate_dropout like this,

```
   def activate_dropout(self):
        # collect all Dropout module
        for module in self.model.modules():
            if isinstance(module, nn.Dropout) or isinstance(module, DropPath):
                module.train()
```

and inference using mmdet apis.

i confirmed that every outputs were changed little bit on the same input.
But, i was curious that MC method is a effective solutions in real problem.

**Comment by jmaxrdgz** [comment]
Hi, seems like a good solution. I ended up creating a custom decoder block with an additional dropout because I wanted to keep the ability to have different training and inference dropout probabilities. 

And to answer your question, I had no success in using the MC method to estimate correctness of DINO's predictions, it gave pretty uncorelated results. Though I had more success with layer-based variances.
